### ISSUE TITLE
Automatically open in container

### ISSUE THREAD

**Issue by turowicz** [microsoft/vscode-remote-release#2900]
Please add a setting for always attempting to open in a container if a project contains the .devcontainer folder.

**Comment by turowicz** [comment]
@Chuxel EXACTLY.

Currently I flip between different projects (8-10 at a time) and each time I have to wait for the popup and click "Reopen in Contanier" manually.

**Comment by yngwi** [comment]
Hi @turowicz , I'm not sure if you know, but on Windows you can at least "pin" container-projects. This way they are immediately openable from the icon on the taskbar and the start menu. Maybe this helps.

**Comment by turowicz** [comment]
I'm on Ubuntu 20.04 and OSX

**Comment by yngwi** [comment]
> 
> 
> @yngwi I'm on Ubuntu 20.04 and OSX

I'm sorry, I didn't realize.

**Comment by HenkVanMaanen** [comment]
Any update on this?

**Comment by turowicz** [comment]
I have switched to "Open Repository in Container". No longer a problem for me.

**Comment by ldiebold** [comment]
@turowicz 

What do you mean by:
> switched to "Open Repository in Container"

Is this a setting? an extension?

**Comment by turowicz** [comment]
@ldiebold 
![image](https://user-images.githubusercontent.com/1055391/109782081-c9134d00-7c08-11eb-9fb0-55ea97d89f46.png)

**Comment by ldiebold** [comment]
@turowicz Ahh I see. Still kind of a workaround. Maybe we can find an extension that allows us to run a command when the editor has loaded to.

**Comment by turowicz** [comment]
@ldiebold if using Docker For Windows, that way the filesystem is much faster and you can get 10x on build times.

**Comment by ldiebold** [comment]
@turowicz I'm on Linux 😊

**Comment by turowicz** [comment]
@ldiebold I'm on Linux to, you can still get 10% off build time.

**Comment by ldiebold** [comment]
@turowicz now I'm just confused. You can get Docker for windows on Linux?

**Comment by turowicz** [comment]
No, you can have more workstations.

**Comment by turowicz** [comment]
LOL

**Comment by jchallenger** [comment]
Would be great if I could open into a devcontainer from the command line, like ```code . --devcontainer```

**Comment by NicholasBallard** [comment]
> Would be great if I could open into a devcontainer from the command line, like `code . --devcontainer`

@jchallenger This is what I came here to say.

**Comment by tobidalhof** [comment]
> Would be great if I could open into a devcontainer from the command line, like `code . --devcontainer`

Yes, please!

**Comment by jrcasso** [comment]
Something like `code . --devcontainer` and a short `code -d .` would be fantastic. The request of this feature support for the CLI is really a testament to the pervasiveness of this development strategy!

**Comment by rhlobo** [comment]
Looking forward for this `code . --devcontainer` option. 

As stated before, something similar was on the #5428 iteration, but it wasn't implemented, and wasn't moved to the next iteration plan (#5552).

**Comment by sjroe** [comment]
> It is already there. You can run:
> 
> ```
> devcontainer open .
> ```

For reference, this requires the [devcontainer-cli](https://code.visualstudio.com/docs/remote/devcontainer-cli)

**Comment by peterswords** [comment]
> I'm not sure if you know, but on Windows you can at least "pin" container-projects. This way they are immediately openable from the icon on the taskbar and the start menu. Maybe this helps.

Fabulous suggestion. (I'm using StartIsBack for Windows 7-style menus). I can pin the dev-container versions of projects:

![image](https://user-images.githubusercontent.com/8174120/142728337-b940a765-2202-44de-825d-f81919ec38bb.png)

**Comment by ssbarnea** [comment]
Does anyone know how to to tell code to open specific folder using wsl **from the command line**? That is essential for testing purposes.

**Comment by peterswords** [comment]
> Does anyone know how to to tell code to open specific folder using wsl **from the command line**? That is essential for testing purposes.

@ssbarnea, you can just cd to the folder and type "code ."

(or presumably code <directory name>)

**Comment by ssbarnea** [comment]
The solution is `code --remote wsl+Ubuntu /mn/c/Users/johndoe/c/myrepo` and would work if you already have the wsl-remote extension installed. The path given must be the Unix one, so you may need to covert Windows one to Linux. Keep in mind that drive letter becomes lowercase but other things keep the case.

**Comment by dlangerm** [comment]
Whatever happened to this? I'm building an internal tool for my team and this would be super handy.

**Comment by gjsjohnmurray** [comment]
See also #2133

**Comment by cswrd** [comment]
Another workaround: ```File > Save Workspace As ...```. On Windows just double click the file. But seems to break if you add additional folders to the workspace.

**Comment by nidrizi** [comment]
If you often use `code .` to open vs code on the current folder you might find this useful:
`vscode --remote .`

**Comment by sebastiaoLa** [comment]
Hey guys, although we can do it using the devcontainers CLI I would like to do it on a more global way, Is there a change we will get a setting for this?

**Comment by amaltson** [comment]
Yes, the hope would be to have a VSCode JSON configuration to have the editor automatically reload if there's a DevContainer or even start up automatically.

**Comment by wparad** [comment]
5 years later, and how is this still not a feature?

**Comment by a-bj-q** [comment]
Because nobody can be bothered.

**Comment by ericguthmann** [comment]
Would very much like this feature.  I'm on MacOS, so pinning is not an option.

**Comment by felipecrs** [comment]
I use the [Project Manager](https://marketplace.visualstudio.com/items?itemName=alefragnani.project-manager) extension. You can save the project when inside the devcontainer, and then Project Manager will always open it in the devcontainer.

**Comment by futurejohn** [comment]
> I use the [Project Manager](https://marketplace.visualstudio.com/items?itemName=alefragnani.project-manager) extension. You can save the project when inside the devcontainer, and then Project Manager will always open it in the devcontainer.

That's working great for me, thank you! I wish there was a way to integrate it into Spotlight or Raycast. Have you found a way to do that?
